util/graph.js

```diff
@@ -0,0 +1,77 @@
+/** Class that signifies a graph */
+class Graph {
+    /**
+      * The underlying graph object. This is an object with nodes and edges.
+      * The class contains an adjacency list property that represents the edges
+     * between each of the nodes. Each edge can have attributes such as weight (which 
+     * by default is 10) and whether or not the edge is directed.
+     * 
+      */
+    adjacencyList;
+ 
+    constructor() {
+       this.adjacencyList = []
+    }
+ 
+    clear() {
+       this.adjacencyList = []
+       return true
+    }
+ 
+    addNode(node) {
+       if (!this.adjacencyList[node]) {
+          this.adjacencyList[node] = []
+       }
+    }
+ 
+    addNodes(nodes) {
+       if (!Array.isArray(nodes)) {
+          return false
+       }
+       nodes.forEach(node => {
+          this.addNode(node)
+       })
+    }
+ 
+    addEdge(node1, node2, weight=10, directed=true) {
+       this.adjacencyList[node1].push({ node: node2, weight })
+       if (!directed) {
+          this.adjacencyList[node2].push({ node: node1, weight })
+       }
+    }
+ 
+    removeEdge(node1, node2) {
+       this.adjacencyList[node1] = this.adjacencyList.filter(
+          node => node !== node1
+       )
+       if (!directed) {
```

Comment:
https://github.com/tair/mb4-service/pull/58/files#r1170655264
"`directed` is not defined here but since this is default directed, you can delete this condition."